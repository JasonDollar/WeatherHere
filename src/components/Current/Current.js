import React from 'react'
import moment from 'moment'
import Icon from '../Icon/Icon'



import classes from './Current.module.scss'



const Current = ({
  currently, daily, hourly, text, locale,
}) => {
  console.log(hourly)
  moment.locale('pl')
  // console.log(locale, currently, moment(1541372400 * 1000).format('LL'))
  // const date = moment(currently.sunriseTime * 1000).format('LL')
  return (
    <div>
      <div className={classes.head}>
        <span>Miasto</span>
        <span>{currently.temperature}</span>
        
        <Icon icon={currently.icon} />
      </div>
      <div>
        <span>Data</span>
        <span>
          {'currently: '}
          {moment(currently.time * 1000).format()}
        </span>
        {' hourtly'}

        <span>{moment(hourly.data[0].time * 1000).format()}</span>
      </div>
      <div>{`${currently.summary} daily`}</div>
      <div>{`${hourly.summary} houtly`}</div>

      <ul>
        <li>{`preciProp ${hourly.data[0].precipProbability}`}</li>
        <li>{`[precip int] ${hourly.data[0].precipIntensity}`}</li>
        <li>{`temp ${hourly.data[0].temperature}`}</li>
        <li>{`apparent temp ${hourly.data[0].apparentTemperature}`}</li>
        <li>{`dewpoint ${hourly.data[0].dewPoint}`}</li>
        <li>{`humidity ${hourly.data[0].humidity}`}</li>
        <li>{`pressure ${hourly.data[0].pressure}`}</li>
        <li>{`windspeed ${hourly.data[0].windSpeed}`}</li>
        <li>{`windgust ${hourly.data[0].windGust}`}</li>
        <li>{`w.bearing ${hourly.data[0].windBearing}`}</li>
        <li>{`cloud cover ${hourly.data[0].cloudCover}`}</li>
        <li>{`visibility ${hourly.data[0].visibility}`}</li>
        <li>{`uvindex ${hourly.data[0].uvIndex}`}</li>
        <li>-----------------------------</li>
        <li>{`sunrise ${moment(daily.data[0].sunriseTime * 1000).format('LTS')}`}</li>
        <li>{`sunset ${moment(daily.data[0].sunsetTime * 1000).format('LTS')}`}</li>
        <li>{`moonphase ${daily.data[0].moonPhase}`}</li>
        <li>{`uv current ${daily.data[0].uvIndex}`}</li>
        <li>{`ozone ${daily.data[0].ozone}`}</li>
        <li>{`precip 5type ${daily.data[0].precipType}`}</li>
      </ul>
      
    </div>
  )
}

export default Current

/*
time
summary

precipPropability
precipIntensity
temperature
apparentTemperature
dewPoint
humidity
pressure
windSpeed
windGust
windBearing (kierunek)
cloudCover
visibility
uvIndex


sunriseTime
sunsetTime
moonPhase
uvIndex
ozone
precipType
summary
*/