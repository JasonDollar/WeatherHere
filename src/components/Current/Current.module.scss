.Current {
  // height: 150px;
  
  & .main {
    font-size: var(--head-font-size);
    display: flex;
    justify-content: space-between;

    & svg {
      height: var(--head-font-size);
      width: var(--head-font-size);
      transform: scale(2);
      vertical-align: middle;
    }
  }

  .circlesContainer {

    & ul {
      display: flex;
      list-style: none;
      justify-content: space-around;

      // & li {
      //   flex: 0 0 25%;
      // }
    }
  }

  &__list {
    display: flex;
    flex-wrap: wrap;
    margin: 3rem 0;
    list-style: none;

    & li {
      flex: 0 0 33.3333%;
    }
  }
}