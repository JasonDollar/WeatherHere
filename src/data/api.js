const DARK_API = 'a077c3a2c30623e41ef4a1cafbec622d'
const DARK_URL = 'https://api.darksky.net/forecast/'

export { DARK_API, DARK_URL }  

