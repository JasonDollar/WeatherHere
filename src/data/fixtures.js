import weatherData from './random'
import languageText from './text-locale' 

export const appTestData = {
  weather: weatherData,
  units: 'si',
  language: 'pl',
  text: languageText,
  timezone: 'Europe/Warsaw',
  locationShortName: 'Plock',
  
}