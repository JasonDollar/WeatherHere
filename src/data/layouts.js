const layouts = {
  pink: {
    id: 'pink',
    name: 'Pink',
    primary: '#dd0055',
    primaryOffset: '#f5689e',
    mobileMenu: '#dd0055',
    antiGraph: '#8884d8',
    secondaryGraph: '#0c1052',
    background: '#fffcfc',
    background2: '#fffcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#eae9e9',
    greyGraph: '#676767',
  },
  pinkDark: {
    id: 'pinkDark',
    name: 'Pink Dark',
    primary: '#dd0055',
    primaryOffset: '#dd0055',
    mobileMenu: '#f5689e',
    antiGraph: '#8884d8',
    secondaryGraph: '#dd0055',
    background: '#272424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#efeee7',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#333',
    greyGraph: '#acacac',
  },
  pumpkin: {
    id: 'pumpkin',
    name: 'Pumpkin Dark',
    primary: '#f48042',
    primaryOffset: '#ef915f',
    mobileMenu: '#f48042',
    antiGraph: '#50c25f',
    secondaryGraph: '#f48042',
    background: '#262424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eee',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  evening: {
    id: 'evening',
    name: 'Evening Dark',
    primary: '#b92b27',
    primaryOffset: '#d84f4b',
    mobileMenu: '#d84f4b',
    antiGraph: '#1565C0',
    secondaryGraph: '#d84f4b',
    background: '#282424',
    background2: '#fcfcfc',
    backgroundMobile: '#333',
    text: '#eeeef8',
    textSecondary: '#ddd',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#999',
    grey3: '#363333',
    greyGraph: '#acacac',
  },
  sky: {
    id: 'sky',
    name: 'Cool Sky',
    primary: '#2980B9',
    primaryOffset: '#6DD5FA',
    mobileMenu: '#2980B9',
    antiGraph: '#756ef0',
    secondaryGraph: '#2980B9',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7e7ec',
    greyGraph: '#676767',
  },
  harvey: {
    id: 'harvey',
    name: 'Harvey',
    primary: '#2b574b',
    primaryOffset: '#75d1a6',
    mobileMenu: '#386e60',
    antiGraph: '#0daeca',
    secondaryGraph: '#2b574b',
    background: '#fcfcfc',
    background2: '#fcfcfc',
    backgroundMobile: '#ddd',
    text: '#222',
    textSecondary: '#eee',
    textTransparent: '#22222280',
    border: '#aaa',
    borderTransparent: '#aaaaaa55',
    boxShadow: '#222',
    grey1: '#888',
    grey2: '#bbb',
    grey3: '#e7eae7',
    greyGraph: '#676767',
  },


}

export default layouts



