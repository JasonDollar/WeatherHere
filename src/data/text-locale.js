const pl = {
  header: {
    title: 'Pogoda',
    search: 'Szukaj',
  },  
  layout: {
    search: 'Szukaj',
  },
  current: {
    today: 'Dziś',
    humidity: 'Wilgotność',
    wind: 'Wiatr',
    precipProb: 'Opady',
    precipType: 'Rodzaj opadu',
    precipInt: 'Intensywność opadu',
    appTemp: 'Temperatura odczuwalna',
    dewPoint: 'Punkt rosy',
    pressure: 'Ciśnienie',
    cloudCov: 'Zachmurzenie',
    visibility: 'Widoczność',
    uvIndex: 'Indeks UV',
    sunrise: 'Wschód słońca',
    sunset: 'Zachód słońca',
    moonphase: 'Faza księżyca',
    ozone: 'Ozon',
    windGust: 'Wiatr w porywach',
    location: 'Nieznana lokacja',
  },
  date: {
    weekDay: ['Niedziela', 'Poniedziałek', 'Wtorek', 'Środa', 'Czwartek', 'Piątek', 'Sobota', 'Dziś'],
    weekDayShort: ['Niedz.', 'Pon.', 'Wt.', 'Śr.', 'Czw.', 'Pt.', 'Sob.'],
    today: 'Dziś',
    temperature: 'Temperatura (°C)',
    temperatureMax: 'Temperatura Max.',
    temperatureMin: 'Temperatura Min.',
  },
  hourly: {
    hour: 'Godzina',
    precipProb: 'Możliwość Opadów (%)',
    precipProbShort: 'Możliwość Opadów',
    temperature: 'Temperatura (°C)',
    temperatureShort: 'Temperatura',
  },
  update: 'Ostatnia aktualizacja: ',
}

const en = {
  header: {
    title: 'Weather',
    
  },
  layout: {
    search: 'Search',
  },
  current: {
    today: 'Today',
    humidity: 'Humidity',
    wind: 'Wind',
    precipProb: 'Precipitation chance',
    precipType: 'Precipitation type',
    precipInt: 'Precipitation intensity',
    appTemp: 'Apparent temperature',
    dewPoint: 'Dew point',
    pressure: 'Pressure',
    cloudCov: 'Cloud cover',
    visibility: 'Visibility',
    uvIndex: 'UV Index',
    sunrise: 'Sunrise',
    sunset: 'Sunset',
    moonphase: 'Moon phase',
    ozone: 'Ozone',
    windGust: 'Wind gust',
    location: 'Unknown location',
  },
  date: {
    weekDay: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
    weekDayShort: ['Sun.', 'Mon.', 'Tue.', 'Wed.', 'Thu.', 'Fri.', 'Sat.'],
    today: 'Today',
    temperature: 'Temperature (°C)',
    temperatureMax: 'Temperature Max.',
    temperatureMin: 'Temperature Min',
  },
  hourly: {
    hour: 'Hour',
    precipProb: 'Precipitation chance (%)',
    precipProbShort: 'Precipitation chance',
    temperature: 'Temperature (°C)',
    temperatureShort: 'Temperature',
  },
  update: 'Last updated: ',
}

export { pl, en } 