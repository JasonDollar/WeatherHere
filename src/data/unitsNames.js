export const unitsNames = {
  si: {
    id: 'si',
    distance: 'km',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'm/s',
    pressure: 'hPa',
  },
  us: {
    id: 'us',
    distance: 'mi',
    precip: 'in/h',
    precipAcc: 'in',
    temperature: '째F',
    wind: 'mi/h',
    pressure: 'hPa',
  },
  uk2: {
    id: 'uk2',
    distance: 'mi',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'mi/h',
    pressure: 'hPa',
  },
  ca: {
    id: 'ca',
    distance: 'km',
    precip: 'mm/h',
    precipAcc: 'cm',
    temperature: '째C',
    wind: 'km/h',
    pressure: 'hPa',
  },
}