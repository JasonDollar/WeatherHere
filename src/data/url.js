const DARK_URL = 'https://api.darksky.net/forecast/'

export { DARK_URL }