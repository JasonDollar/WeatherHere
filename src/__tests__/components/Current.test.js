import React from 'react'
import { shallow } from 'enzyme'
import { appTestData as data } from '../../data/fixtures'
import Current from '../../components/Current/Current'



let dateNowSpy

beforeAll(() => {
  // Lock Time
  dateNowSpy = jest.spyOn(Date, 'now').mockImplementation(() => 1000000)
})
describe('Current component tests', () => {
  test('render Current component', () => {
    const wrapper = shallow(
      <Current 
        currently={data.weather.currently} 
        daily={data.weather.daily} 
        currentText={data.text.pl.current} 
        dateText={data.text.pl.date}
        timezone={data.timezone} 
        locationShortName={data.locationShortName} 
        units={data.units.si.id}
      />,
    )
 
    expect(wrapper).toMatchSnapshot()
  })
})

afterAll(() => {
  // Unlock Time
  dateNowSpy.mockRestore()
})