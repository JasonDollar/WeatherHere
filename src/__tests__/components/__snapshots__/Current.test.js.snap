// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current component tests Matches Snapshot 1`] = `
<Fragment>
  <MainInfo
    currentText={
      Object {
        "appTemp": "Temperatura odczuwalna",
        "cloudCov": "Zachmurzenie",
        "date": "Data",
        "dewPoint": "Punkt rosy",
        "humidity": "Wilgotność",
        "icon": "Ikona pogody",
        "location": "Nieznana lokacja",
        "moon": Array [
          "Nów",
          "Wschodzący sierp",
          "Pierwsza kwadra",
          "Wschodzący księżyc",
          "Pełnia",
          "Ubywający księżyc",
          "Trzecia kwadra",
          "Ubywający sierp",
        ],
        "moonphase": "Faza księżyca",
        "ozone": "Ozon",
        "precipInt": "Intensywność opadu",
        "precipProb": "Opady",
        "precipType": "Rodzaj opadu",
        "pressure": "Ciśnienie",
        "sun": "Słońce",
        "sunrise": "Wschód słońca",
        "sunset": "Zachód słońca",
        "today": "Dziś",
        "uvIndex": "Indeks UV",
        "visibility": "Widoczność",
        "wind": "Wiatr",
        "windGust": "Wiatr w porywach",
      }
    }
    currently={
      Object {
        "apparentTemperature": 6.53,
        "cloudCover": 0.92,
        "dewPoint": 8.45,
        "humidity": 0.94,
        "icon": "partly-cloudy-night",
        "ozone": 290.56,
        "precipIntensity": 0.0076,
        "precipProbability": 0.01,
        "precipType": "rain",
        "pressure": 1009.97,
        "summary": "Średnie zachmurzenie",
        "temperature": 9.38,
        "time": 1540834485,
        "uvIndex": 0,
        "visibility": 16.09,
        "windBearing": 99,
        "windGust": 10.76,
        "windSpeed": 5.76,
      }
    }
    dateText={
      Object {
        "temperature": "Temperatura",
        "temperatureMax": "Temperatura Max.",
        "temperatureMin": "Temperatura Min.",
        "today": "Dziś",
        "weekDay": Array [
          "Niedziela",
          "Poniedziałek",
          "Wtorek",
          "Środa",
          "Czwartek",
          "Piątek",
          "Sobota",
          "Dziś",
        ],
        "weekDayShort": Array [
          "Nie",
          "Pon",
          "Wt",
          "Śr",
          "Czw",
          "Pt",
          "Sob",
        ],
      }
    }
    locationShortName="Plock"
    sunPositionTime={
      Object {
        "sunrise": 1540791182,
        "sunset": 1540826320,
      }
    }
    timezone="Europe/Warsaw"
    today={
      Object {
        "day": "4",
        "hour": "01:16",
        "month": "1.1",
      }
    }
    units="si"
  />
  <CircleContainer
    currentText={
      Object {
        "appTemp": "Temperatura odczuwalna",
        "cloudCov": "Zachmurzenie",
        "date": "Data",
        "dewPoint": "Punkt rosy",
        "humidity": "Wilgotność",
        "icon": "Ikona pogody",
        "location": "Nieznana lokacja",
        "moon": Array [
          "Nów",
          "Wschodzący sierp",
          "Pierwsza kwadra",
          "Wschodzący księżyc",
          "Pełnia",
          "Ubywający księżyc",
          "Trzecia kwadra",
          "Ubywający sierp",
        ],
        "moonphase": "Faza księżyca",
        "ozone": "Ozon",
        "precipInt": "Intensywność opadu",
        "precipProb": "Opady",
        "precipType": "Rodzaj opadu",
        "pressure": "Ciśnienie",
        "sun": "Słońce",
        "sunrise": "Wschód słońca",
        "sunset": "Zachód słońca",
        "today": "Dziś",
        "uvIndex": "Indeks UV",
        "visibility": "Widoczność",
        "wind": "Wiatr",
        "windGust": "Wiatr w porywach",
      }
    }
    currently={
      Object {
        "apparentTemperature": 6.53,
        "cloudCover": 0.92,
        "dewPoint": 8.45,
        "humidity": 0.94,
        "icon": "partly-cloudy-night",
        "ozone": 290.56,
        "precipIntensity": 0.0076,
        "precipProbability": 0.01,
        "precipType": "rain",
        "pressure": 1009.97,
        "summary": "Średnie zachmurzenie",
        "temperature": 9.38,
        "time": 1540834485,
        "uvIndex": 0,
        "visibility": 16.09,
        "windBearing": 99,
        "windGust": 10.76,
        "windSpeed": 5.76,
      }
    }
    units="si"
  />
  <ListInfo
    currentText={
      Object {
        "appTemp": "Temperatura odczuwalna",
        "cloudCov": "Zachmurzenie",
        "date": "Data",
        "dewPoint": "Punkt rosy",
        "humidity": "Wilgotność",
        "icon": "Ikona pogody",
        "location": "Nieznana lokacja",
        "moon": Array [
          "Nów",
          "Wschodzący sierp",
          "Pierwsza kwadra",
          "Wschodzący księżyc",
          "Pełnia",
          "Ubywający księżyc",
          "Trzecia kwadra",
          "Ubywający sierp",
        ],
        "moonphase": "Faza księżyca",
        "ozone": "Ozon",
        "precipInt": "Intensywność opadu",
        "precipProb": "Opady",
        "precipType": "Rodzaj opadu",
        "pressure": "Ciśnienie",
        "sun": "Słońce",
        "sunrise": "Wschód słońca",
        "sunset": "Zachód słońca",
        "today": "Dziś",
        "uvIndex": "Indeks UV",
        "visibility": "Widoczność",
        "wind": "Wiatr",
        "windGust": "Wiatr w porywach",
      }
    }
    currently={
      Object {
        "apparentTemperature": 6.53,
        "cloudCover": 0.92,
        "dewPoint": 8.45,
        "humidity": 0.94,
        "icon": "partly-cloudy-night",
        "ozone": 290.56,
        "precipIntensity": 0.0076,
        "precipProbability": 0.01,
        "precipType": "rain",
        "pressure": 1009.97,
        "summary": "Średnie zachmurzenie",
        "temperature": 9.38,
        "time": 1540834485,
        "uvIndex": 0,
        "visibility": 16.09,
        "windBearing": 99,
        "windGust": 10.76,
        "windSpeed": 5.76,
      }
    }
    moonphase={0.67}
    units="si"
  />
</Fragment>
`;
